praw/models/reddit/widgets.py

```diff
@@ -0,0 +1,533 @@
+"""Provide classes relating to widgets."""
+
+from ...const import API_PATH
+from ..base import PRAWBase
+
+
+class Button(PRAWBase):
+    """Class to represent a single button inside a :class:`.ButtonWidget`."""
+
+
+class Image(PRAWBase):
+    """Class to represent an image that's part of a :class:`.ImageWidget`."""
+
+
+class ImageData(PRAWBase):
+    """Class for image data that's part of a :class:`.CustomWidget`."""
+
+
+class MenuLink(PRAWBase):
+    """Class to represent a single link inside a menu or submenu."""
+
+
+class Submenu(PRAWBase):
```

Comment:
Can you use `BaseList` as the super class? See `RedditorList` for a simple example.